api/evaluation/evaluation.proto

```diff
@@ -0,0 +1,145 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *           $$\                           $$\ $$\   $$\
+ *           $$ |                          $$ |\__|  $$ |
+ *  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+ * $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+ * $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+ * $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+ * \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+ *  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+ *
+ * This file is part of Clouditor Community Edition.
+ */
+ syntax = "proto3";
+
+ package clouditor.evaluation.v1;
+
+ import "google/api/annotations.proto";
+ import "google/protobuf/timestamp.proto";
+ import "api/assessment/assessment.proto";
+ import "api/orchestrator/orchestrator.proto";
+ import "tagger/tagger.proto";
+ import "validate/validate.proto";
+
+ option go_package = "clouditor.io/clouditor/api/evaluation";
+
+ // This service represents the Evaluation Interface
+ service Evaluation {
+    // StartEvaluation evaluates all assessment results of an service based on its Target of Evaluation (binding of a cloud service to a catalog). The assessment results are evaluated regarding the contol ID.
+    rpc StartEvaluation(StartEvaluationRequest) returns (StartEvaluationResponse) {
+    option (google.api.http) = {
+        post : "/v1/evaluation/evaluate/start"
+        body : "*"
+      };
+     } 
+
+    // StopEvaluation stop the evaluation for the given Cloud Service
+    rpc StopEvaluation(StopEvaluationRequest) returns (StopEvaluationResponse) {
+    option (google.api.http) = {
+        post : "/v1/evaluation/evaluate/{category_name}/{control_id}/stop"
+        body : "target_of_evaluation"
+        };
+    } 
+
+    // List all evaluation results. Part of the public API, also exposed as REST.
+    rpc ListEvaluationResults(ListEvaluationResultsRequest)
+        returns (ListEvaluationResultsResponse) {
+        option (google.api.http) = {
+            get : "/v1/evaluation/results"
+        };
+    }
+}
+
+message ListEvaluationResultsRequest {
+    optional string filtered_cloud_service_id = 1
+        [(validate.rules).string.min_len = 1];
+  
+    optional string filtered_control_id = 2
+        [(validate.rules).string.min_len = 1];
+  
+    int32 page_size = 10;
+    string page_token = 11;
+    string order_by = 12;
+    bool asc = 13;
+  }
+  message ListEvaluationResultsResponse {
+    repeated EvaluationResult results = 1;
+    string next_page_token = 2;
+  }
+
+message StartEvaluationRequest {
+    clouditor.orchestrator.v1.TargetOfEvaluation target_of_evaluation = 1
```

Comment:
do we really want the complete message here and not just the service ID and the catalog ID?

Reply:
You are right, after the other changes, we can get the ToE from the orchestrator. :+1: